### PR TITLE
fix: typeCast string() parameters for JSON

### PR DESCRIPTION
**Pull request by jyoonPro**
#1661

### CODE REVIEW

**Comment by sidorares** [comment]
would you be able to add a simple unit test, ideally in the context of json parsing ( e.i: create temp table with json column, insert data, read it using typeCast ) The test need to fail without proposed change to `lib/parsers/text_parser.js`

The change itself looks good to me. ( I wonder if we should be more helpful proactively? A warning in typeCast if the `string()` is used without charset AND field type is JSON )

Also could you try to follow https://www.conventionalcommits.org/en/v1.0.0/ in commit messages @jyoonPro?

**Comment by jyoonPro** [comment]
@sidorares How should we warn the user? I'm not sure if I should use console.log() since no other part of the code seems to be using it.

**Comment by sidorares** [comment]
> @sidorares How should we warn the user? I'm not sure if I should use console.log() since no other part of the code seems to be using it.

This is how we warn when the user `await`s non-promise query ( also this is immediately followed by `throw` so maybe not the best example )
https://github.com/sidorares/node-mysql2/blob/241cb10658aa32b410d4f4a3b618156e8bec6ff5/lib/commands/query.js#L42

File: lib/parsers/text_parser.js [hunk 1]
```diff
@@ -85,8 +85,15 @@ function compile(fields, options, config) {
       db: field.schema,
       table: field.table,
       name: field.name,
-      string: function() {
-        return _this.packet.readLengthCodedString(field.encoding);
+      string: function(encoding = field.encoding) {
+        if (field.columnType === Types.JSON && encoding === field.encoding) {
```
Comment (Marsup) [inline lib/parsers/text_parser.js:89]: Shouldn't you also check that encoding is not already utf8? I have a case where `field.encoding` is, and passing utf8 explicitly still raises the warning.

**Pull request merged**